cmd/ctl/pkg/status/certificate/types.go

```diff
@@ -442,3 +482,115 @@ func (orderStatus *OrderStatus) String() string {
+		reflect.DeepEqual(x.CreationTime, y.CreationTime) && reflect.DeepEqual(x.Conditions, y.Conditions) &&
+		reflect.DeepEqual(x.DNSNames, y.DNSNames) && reflect.DeepEqual(x.Events, y.Events) &&
+		reflect.DeepEqual(x.NotBefore, y.NotBefore) && reflect.DeepEqual(x.NotAfter, y.NotAfter) &&
+		reflect.DeepEqual(x.RenewalTime, y.RenewalTime) &&
+		IssuerStatusDeepEqual(x.IssuerStatus, y.IssuerStatus) &&
+		SecretStatusDeepEqual(x.SecretStatus, y.SecretStatus) &&
+		CRStatusDeepEqual(x.CRStatus, y.CRStatus) &&
+		OrderStatusDeepEqual(x.OrderStatus, y.OrderStatus) &&
+		ChallengeStatusListDeepEqual(x.ChallengeStatusList, y.ChallengeStatusList)
+
+}
+
+func IssuerStatusDeepEqual(x, y *IssuerStatus) bool {
+	if x == nil || y == nil {
+		return x == y
+	}
+
+	return ((x.Error == nil && y.Error == nil) || x.Error.Error() == y.Error.Error()) &&
+		x.Name == y.Name && x.Kind == y.Kind && reflect.DeepEqual(x.Conditions, y.Conditions)
+}
+
+func SecretStatusDeepEqual(x, y *SecretStatus) bool {
+	if x == nil || y == nil {
+		return x == y
+	}
+
+	return ((x.Error == nil && y.Error == nil) || x.Error.Error() == y.Error.Error()) &&
+		x.Name == y.Name && reflect.DeepEqual(x.IssuerCountry, y.IssuerCountry) &&
+		reflect.DeepEqual(x.IssuerOrganisation, y.IssuerOrganisation) && x.IssuerCommonName == y.IssuerCommonName &&
+		reflect.DeepEqual(x.KeyUsage, y.KeyUsage) && reflect.DeepEqual(x.ExtKeyUsage, y.ExtKeyUsage) &&
+		reflect.DeepEqual(x.PublicKeyAlgorithm, y.PublicKeyAlgorithm) && reflect.DeepEqual(x.SignatureAlgorithm, y.SignatureAlgorithm) &&
+		reflect.DeepEqual(x.SubjectKeyId, y.SubjectKeyId) && reflect.DeepEqual(x.AuthorityKeyId, y.AuthorityKeyId) &&
+		reflect.DeepEqual(x.SerialNumber, y.SerialNumber)
+}
+
+func CRStatusDeepEqual(x, y *CRStatus) bool {
+	if x == nil || y == nil {
+		return x == y
+	}
+
+	return ((x.Error == nil && y.Error == nil) || x.Error.Error() == y.Error.Error()) &&
+		x.Name == y.Name && x.Namespace == y.Namespace &&
+		reflect.DeepEqual(x.Conditions, y.Conditions) && reflect.DeepEqual(x.Events, y.Events)
+}
+
+func OrderStatusDeepEqual(x, y *OrderStatus) bool {
+	if x == nil || y == nil {
+		return x == y
+	}
+
+	return ((x.Error == nil && y.Error == nil) || x.Error.Error() == y.Error.Error()) &&
+		x.Name == y.Name && x.State == y.State && x.Reason == y.Reason &&
+		reflect.DeepEqual(x.Authorizations, y.Authorizations) && reflect.DeepEqual(x.FailureTime, y.FailureTime)
+}
+
+func ChallengeStatusListDeepEqual(x, y *ChallengeStatusList) bool {
+	if x == nil || y == nil {
+		return x == y
+	}
+
+	if !((x.Error == nil && y.Error == nil) || x.Error.Error() == y.Error.Error()) {
+		return false
+	}
+
+	if len(x.ChallengeStatuses) != len(y.ChallengeStatuses) {
+		return false
+	}
+
+	for i := range x.ChallengeStatuses {
+		if !ChallengeStatusDeepEqual(x.ChallengeStatuses[i], y.ChallengeStatuses[i]) {
+			return false
+		}
+	}
+	return true
+}
+
+func ChallengeStatusDeepEqual(x, y *ChallengeStatus) bool {
```

Comment:
What is the advantage of these over `reflect.DeepEqual`?

Reply:
I thought they were necessary because bazel doesn't like using on `reflect.DeepEqual` on `error`s, but it turns out we can just use `assert.Equals` and bazel is happy. I have removed these functions now.